recipes-browser/wpewebkit/wpewebkit.inc

```diff
@@ -85,6 +85,7 @@ PACKAGECONFIG[mediasource] = "-DENABLE_MEDIA_SOURCE=ON,-DENABLE_MEDIA_SOURCE=OFF
 PACKAGECONFIG[mediastream] = "-DENABLE_MEDIA_STREAM=ON -DUSE_GSTREAMER_TRANSCODER=OFF,-DENABLE_MEDIA_STREAM=OFF,gstreamer1.0 gstreamer1.0-plugins-bad"
 PACKAGECONFIG[service-worker] = "-DENABLE_SERVICE_WORKER=ON,-DENABLE_SERVICE_WORKER=OFF,"
 PACKAGECONFIG[soup2] = "-DUSE_SOUP2=ON,-DUSE_SOUP2=OFF,libsoup-2.4"
+PACKAGECONFIG[touch-slider] = "-DENABLE_TOUCH_SLIDER=ON,-DENABLE_TOUCH_SLIDER=OFF,"
```

Comment:
Hi @colin-pm, apparently this option is always active: In `cmake/WebKitFeatures.cmake`, the `ENABLE_TOUCH_SLIDER` is activated through `ENABLE_TOUCH_EVENTS`:

`WEBKIT_OPTION_DEPEND(ENABLE_TOUCH_SLIDER ENABLE_TOUCH_EVENTS)`

; and `ENABLE_TOUCH_EVENTS` is already activated by default `WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_TOUCH_EVENTS PRIVATE ON)` `cmake/OptionsWPE.cmake`.

For the same purpose. It could be also added another flag for `touch-events`

 ```suggestion
PACKAGECONFIG[touch-slider] = "-DENABLE_TOUCH_SLIDER=ON,-DENABLE_TOUCH_SLIDER=OFF,"
PACKAGECONFIG[touch-events] = "-DENABLE_TOUCH_EVENTS=ON,-DENABLE_TOUCH_EVENTS=OFF,"
```

Both my concern to accept this change is because these 2 options are defined as private for the project.

What is the idea? Are you interested on disabling those flags?

Reply:
In general we recommend against toggling private options WebKit port options. If enabling or disabling `ENABLE_TOUCH_SLIDER` fixes an issues, we should consider applying the change unconditionally in the WebKit source tree.

Reply:
@pabelenda I'm fairly confident ENABLE_TOUCH_SLIDER is not enabled by default. My understanding is WEBKIT_OPTION_DEPEND ensures ENABLE_TOUCH_SLIDER cannot be enabled without ENABLE_TOUCH_EVENTS enabled, but does not actually enable the option.

Adding this PACKAGECONFIG option through a bbappends fixed an issue I was seeing where interacting with HTML5 range inputs was not working with a touch screen until I added this PACKAGECONFIG option and enabled it.

I think it wouldn't be a bad idea to add an option of touch-events as well and would accept the commit suggestion.

@aperezdc If the WebKit source tree is updated to enabled ENABLE_TOUCH_SLIDER if ENABLE_TOUCH_EVENTS is enabled, that's fine. However that will require me to add a patch for the WebKit version in 3.1 or go in another direction and add a bbappends to add these PACKAGECONFIG options.

Reply:
> @pabelenda I'm fairly confident ENABLE_TOUCH_SLIDER is not enabled by default. My understanding is WEBKIT_OPTION_DEPEND ensures ENABLE_TOUCH_SLIDER cannot be enabled without ENABLE_TOUCH_EVENTS enabled, but does not actually enable the option.
> 

mmm, right. Good catch :+1: 

I think we should fix this upstream and maybe enable this option by default for WPE or even remove the option completely and enable it unconditionally when touch_events is enabled

Digging into the WebKit git logs seems this feature was added long time ago, in https://bugs.webkit.org/show_bug.cgi?id=88807

> Adding this PACKAGECONFIG option through a bbappends fixed an issue I was seeing where interacting with HTML5 range inputs was not working with a touch screen until I added this PACKAGECONFIG option and enabled it.
> 
> I think it wouldn't be a bad idea to add an option of touch-events as well and would accept the commit suggestion.
> 
> @aperezdc If the WebKit source tree is updated to enable ENABLE_TOUCH_SLIDER if ENABLE_TOUCH_EVENTS is enabled, that's fine. However that will require me to add a patch for the WebKit version in Dunfell or go in another direction and add a bbappends to add these PACKAGECONFIG options.

Are you using the version of WPEWebKit from the master branch (2.38.X) or the old one that is on the dunfell branch (2.32.1) ? If you are using the old one have you tried to use the newer version from master?

Reply:
> > @pabelenda I'm fairly confident ENABLE_TOUCH_SLIDER is not enabled by default. My understanding is WEBKIT_OPTION_DEPEND ensures ENABLE_TOUCH_SLIDER cannot be enabled without ENABLE_TOUCH_EVENTS enabled, but does not actually enable the option.
> 
> mmm, right. Good catch +1
> 
> I think we should fix this upstream and maybe enable this option by default for WPE or even remove the option completely and enable it unconditionally when touch_events is enabled
> 
> Digging into the WebKit git logs seems this feature was added long time ago, in https://bugs.webkit.org/show_bug.cgi?id=88807
> 
After a quick read of the bug comments it seems to me that removing the option and enabling it unconditionally with `ENABLE_TOUCH_EVENTS` is the better option, because it seems that the reason to make the option a conditional back then was:

> I think the patch looks fine, but I would like a way to opt-out it (maybe a USE macro is an overkill?), so BlackBerry port have time to adjust to the implications of this change.

:man_shrugging:

Reply:
> ~~@pabelenda~~ @psaavedra  I'm fairly confident ENABLE_TOUCH_SLIDER is not enabled by default. My understanding is WEBKIT_OPTION_DEPEND ensures ENABLE_TOUCH_SLIDER cannot be enabled without ENABLE_TOUCH_EVENTS enabled, but does not actually enable the option.

Ok. You are right. Anyhow, as it was exposed by @aperezdc in https://github.com/Igalia/meta-webkit/pull/451#discussion_r1129500145, we should avoid to expose developer settings as packageconfig flags. In other hand I am favorable to _enabling it unconditionally_ in WebKit like @clopez  suggest in https://github.com/Igalia/meta-webkit/pull/451#discussion_r1130201206

Reply:
The activation of the TOUCH_SLIDER have implications only in the RangeInputType::handleTouchEvent():

```
#if ENABLE(TOUCH_EVENTS)
void RangeInputType::handleTouchEvent(TouchEvent& event)
{
...
#elif ENABLE(TOUCH_SLIDER)

    if (element()->isDisabledFormControl())
        return;

    if (event.type() == eventNames().touchendEvent) {
        event.setDefaultHandled();
        return;
    }

    RefPtr<TouchList> touches = event.targetTouches();
    if (touches->length() == 1) {
        typedSliderThumbElement().setPositionFromPoint(touches->item(0)->absoluteLocation());
        event.setDefaultHandled();
    }
#else
```

I can't foresee any reason to keep this feature optional and not activated by default.

Reply:
Created this temptative: https://github.com/psaavedra/WebKit/pull/new/eng/Remove-TOUCH_SLIDER-private-CMake-option

Reply:
now that https://github.com/WebKit/WebKit/pull/11288  is merged and  ENABLE_TOUCH_EVENTS privated but set to ON for GTK and WPE:

```
Source/cmake/OptionsGTK.cmake:WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_TOUCH_EVENTS PUBLIC ON)
Source/cmake/OptionsWPE.cmake:WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_TOUCH_EVENTS PRIVATE ON)
```

; I'd suggest to go with the suggestion from https://github.com/Igalia/meta-webkit/pull/451#discussion_r1130916272  for both WebKitGTK and WPEWebKit recipes

Reply:
WDYT @colin-pm. Could you update the MR according this ^?

Reply:
@psaavedra Sounds good to me, thanks for pushing that patch up to WebKit! I updated the PR per your suggestions.